h2o-core/src/main/java/water/FailedNodeWatchdogExtension.java

```diff
@@ -221,7 +194,18 @@ public void run() {
             while (true) {
                 for(H2ONode client: H2O.getClients()){
                     if(isTimeoutExceeded(client, watchdogClientRetryTimeout)){
-                        handleClientDisconnect(client, watchdogClientRetryTimeout);
+                        // if timeout exceeded, check if the client is disconnected
```

Comment:
In watchdog client mode, we additionally need to check that all nodes agree on the fact that the client was disconected, however, the check mechanism whether the client disappeared from the single node needs to be performed also in the non-watchdog mode to handle the client disconnections properly consistently.